Pose estimate

dweizzz opened the pull request:
see commit message for update information
implement pose detection for getting player limb angles in each frame

dweizzz committed:
create new branch to implement pose estimation

kchiem12 committed:
Ensure ffmpeg dependency is installed

- to ensure ffmpeg is installed for macos when reencoding video

kchiem12 committed:
Implement multithreading for object tracking

- Create two processes to track person and ball separately
- Commented lines of code for multithreading

dweizzz committed:
Pose Estimation

- Create branch
- Write src/pose_estimation/pose_estimate.py
- Works, but does not do anything right now
- When running, terminal outputs non fatal errors (?), need to test

dweizzz committed:
Fix ignored files, print angles

dweizzz committed:
Pose Estimation

- Update requirements for Pose Estimation
- Add comments
- Clean code (get rid of prints, format output json)

Mikonooooo committed:
Add ShotAttempt class in state
- ShotAttempt tracks start/end frame, player_id, and ShotType
- did not change any functional code

Mikonooooo committed:
Update ball state tracking for multiple balls
- second parse ball function in general.detect
- states now contains "balls" : {ball[id] : BallState}

Mikonooooo committed:
Implement PlayerState and Frame
- states represented as dictionary of Frames
- Frame represented as object of players and balls dictionaries
- code not yet tested

Mikonooooo committed:
Update SORT parsing with new format
- separate parsing outputs into GameState into parse.py
- update frame-by-frame info on players, balls, and rim
- not tested yet

Mikonooooo committed:
Moved filtering from team_detect to parse as cleaner
- track universal stats in players and balls dictionaries
- not tested yet

bzhang1945 committed:
ci testing

dweizzz committed:
Pose Estimation

- Changed to predict on whole video, process entire video
- Created Pose function in modelrunner and call in main (runs results once)
- Add trained model (best.pt)

Mikonooooo committed:
Saving some stuff

Mikonooooo reviewed (changes requested):
Fix keypoint output in pose estimate
Move pose and weights into own folder

bzhang1945 committed:
shot detect and singleton ball fix

dweizzz committed:
slight alteration

Mikonooooo committed:
Moved frame creation, counting, filtering into state

bzhang1945 committed:
reencoding fix + minor changes

Mikonooooo committed:
Reworked possession processing for state

Mikonooooo committed:
Reworked team detect
- apply spectral clustering to solve sparest cut
- separate possession and pass calculation out for orthogonality

Mikonooooo committed:
Removed general detect, Reason: Obsolete man

Mikonooooo committed:
Reworked shot detection for new state
- with sliding window for detection

Mikonooooo committed:
Fix bugs, app now running

Mikonooooo committed:
Fix court truth map fix

Mikonooooo committed:
Update README to simpler command

Mikonooooo committed:
Update ci.yml

- adjusted ci to run processing

Mikonooooo committed:
Adjusted main to accept command-line arg

audreywangg committed:
Add naive strategy for shot detection

- checks if left wrist above left shoulder and right wrist above right shoulder
- also counts some non-shots (ex: rebounds)
- writes frame of attempted shot and person in tmp/shots.txt

Mikonooooo committed:
Adjusted training_data, added short video to data

Mikonooooo committed:
Merge branch 'ball-202' into pose-estimate
